backend/de.metas.manufacturing/src/main/java/org/eevolution/model/validator/PP_Product_BOMLine.java

```diff
@@ -127,16 +129,19 @@ private boolean noAccordinglyVariantGroup(final I_PP_Product_BOMLine bomLine)
 		return true;
 	}
 
-	@ModelChange(timings = { ModelValidator.TYPE_AFTER_NEW, ModelValidator.TYPE_AFTER_CHANGE, ModelValidator.TYPE_AFTER_DELETE }, skipIfCopying=true)
-	public void updateProductLowestLevelCode(final I_PP_Product_BOMLine bomLine)
+	@ModelChange(timings = { ModelValidator.TYPE_AFTER_NEW, ModelValidator.TYPE_AFTER_CHANGE }, skipIfCopying = true)
+	public void checkingBOMCycle(final I_PP_Product_BOMLine bomLine)
 	{
 		final ProductId productId = ProductId.ofRepoId(bomLine.getM_Product_ID());
 		final I_M_Product product = Services.get(IProductBL.class).getById(productId);
-		final int lowLevel = Services.get(IProductBOMBL.class).calculateProductLowestLevel(ProductId.ofRepoId(product.getM_Product_ID()));
-
-		product.setLowLevel(lowLevel); // update lowlevel
-		InterfaceWrapperHelper.save(product);
+		try
+		{
+			Services.get(IProductBOMBL.class).createParentProductNodeForBOMLine(bomLine);
+		}
+		catch (final BOMCycleException e)
+		{
+			throw new LiberoException("Cycle detected in BOM for product: " + product.getValue());
```

Comment:
pls introduce a user-friendly error (trl to german)

Reply:
also consider throwing this exception  from ProductBOMCycleDetection

Reply:
done